sparkle/src/components/FeedbackSelector.tsx

```diff
@@ -111,47 +174,56 @@ export function FeedbackSelector({
             />
           </div>
         </PopoverTrigger>
-        <PopoverContent fullWidth={true}>
+        <PopoverContent
+          fullWidth={true}
+          onInteractOutside={closePopover}
+          onEscapeKeyDown={closePopover}
+        >
           {isSubmittingThumb ? (
             <div className="m-3 s-flex s-items-center s-justify-center">
               <Spinner size="sm" />
             </div>
           ) : (
             <div className="s-w-80 s-p-4">
               <Page.H variant="h6">
-                {feedback?.thumb === "up"
+                {lastSelectedThumb === "up"
                   ? "🎉 Glad you liked it! Tell us more?"
                   : "🫠 Help make the answers better!"}
               </Page.H>
               <TextArea
                 placeholder={
-                  feedback?.thumb === "up"
+                  lastSelectedThumb === "up"
                     ? "What did you like?"
                     : "Tell us what went wrong so we can make this assistant better."
                 }
-                className="s-mt-4"
+                className="s-mb-4 s-mt-4"
                 rows={3}
                 value={localFeedbackContent ?? ""}
-                onChange={(e) => setLocalFeedbackContent(e.target.value)}
+                onChange={handleTextAreaChange}
               />
+
               {popOverInfo}
-              <div className="s-mt-4 s-flex s-justify-between s-gap-2">
-                <Button
-                  variant="ghost"
-                  label="Skip"
-                  onClick={() => setIsPopoverOpen(false)}
+              <div className="s-mt-2 s-flex s-items-center s-gap-2">
+                <Checkbox
+                  checked={isConversationShared}
+                  onCheckedChange={(value) => {
+                    setIsConversationShared(!!value);
+                  }}
                 />
+                <Page.P variant="secondary">
+                  By clicking, you accept to share your full conversation
```

Comment:
Not very convince by this copy. Shouldn't we surface who will be able to see it if shared?

Reply:
Ah, this one is tricky.
Basically, in the upcoming implementation, (almost) anyone can edit the assistant and see your feedback + the link that opens your conversation.
So We are being broad on purpose. We do display the name of the current builder (last author) in the PopoverContent though.